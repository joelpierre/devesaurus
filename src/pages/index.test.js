import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PureIndex, { Index } from './index';
import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../utils/test-utilities';

const defaultProps = {
  onGetSiteMeta: () => {},
  onGetSiteOptions: () => {},
};

/**
 * Factory function to create a ShallowWrapper for the Index component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<Index {...setupProps} />);
};

describe('<Index/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the Index Component without errors', () => {
    const component = findByTestAttr(wrapper, 'index-main');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(Index, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should get siteMeta and siteOptions on componentDidMount', () => {
    wrapper = setup({
      onGetSiteMeta: jest.fn(),
      onGetSiteOptions: jest.fn(),
    });

    const spy1 = wrapper.instance().props.onGetSiteMeta;
    const spy2 = wrapper.instance().props.onGetSiteOptions;

    wrapper.instance().componentDidMount();

    expect(spy1).toHaveBeenCalled();
    expect(spy2).toHaveBeenCalled();
  });
});
