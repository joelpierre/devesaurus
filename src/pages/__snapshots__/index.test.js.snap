// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index/> should match snapshot 1`] = `
<div
  data-test="component-index"
>
  <main
    className="primary-main primary-main--home"
    data-test="index-main"
  >
    <HeroSearch
      title={null}
    />
    <FeaturedWords
      contrast={true}
      words={
        Array [
          Object {
            "acf": Object {
              "definition": "<p>The art or skill of combining text and pictures in advertisements, magazines, or books.</p>
",
              "examples": false,
              "origin": Object {
                "label": "United Kingdom",
                "value": "GB",
              },
              "part_of_speech": "noun",
              "pronunciation": "graphic-de-sign",
              "syllables": Object {
                "count": "4",
                "list": Array [
                  Object {
                    "item": "gra",
                  },
                  Object {
                    "item": "phic",
                  },
                  Object {
                    "item": "de",
                  },
                  Object {
                    "item": "sign",
                  },
                ],
              },
              "synonyms": Array [
                Object {
                  "post_name": "web-development",
                  "post_status": "publish",
                  "post_title": "Web Development",
                },
              ],
            },
            "slug": "graphic-design",
            "terms": Array [
              Object {
                "name": "Design",
                "slug": "design",
                "taxonomy": "word_tag",
              },
              Object {
                "name": "Graphic Design",
                "slug": "graphic-design",
                "taxonomy": "word_category",
              },
            ],
            "title": "Graphic Design",
          },
          Object {
            "acf": Object {
              "definition": "<p>Web development is the work involved in developing a web site for the Internet (World Wide Web) or an intranet (a private network). Among web professionals, \\"web development\\" usually refers to the main non-design aspects of building web sites: writing markup and coding</p>
",
              "examples": Array [
                Object {
                  "code": "<p><code></p>
<header>
test<br />
</header>
<p></code></p>
",
                  "codepen_url": "",
                  "type": "code-block",
                },
              ],
              "origin": Object {
                "label": "United Kingdom",
                "value": "GB",
              },
              "part_of_speech": "noun",
              "pronunciation": "web~develop~ment",
              "syllables": Object {
                "count": "5",
                "list": Array [
                  Object {
                    "item": "web",
                  },
                  Object {
                    "item": "de",
                  },
                  Object {
                    "item": "ve",
                  },
                  Object {
                    "item": "lop",
                  },
                  Object {
                    "item": "ment",
                  },
                ],
              },
              "synonyms": Array [
                Object {
                  "post_name": "graphic-design",
                  "post_status": "publish",
                  "post_title": "Graphic Design",
                },
              ],
            },
            "slug": "web-development",
            "terms": Array [
              Object {
                "name": "CSS",
                "slug": "css",
                "taxonomy": "word_tag",
              },
              Object {
                "name": "HTML5",
                "slug": "html5",
                "taxonomy": "word_tag",
              },
              Object {
                "name": "Web Development",
                "slug": "web-development",
                "taxonomy": "word_category",
              },
            ],
            "title": "Web Development",
          },
          Object {
            "acf": Object {
              "definition": "<p>Web development is the work involved in developing a web site for the Internet (World Wide Web) or an intranet (a private network). Among web professionals, \\"web development\\" usually refers to the main non-design aspects of building web sites: writing markup and coding</p>
",
              "examples": Array [
                Object {
                  "code": "<p><code></p>
<header>
test<br />
</header>
<p></code></p>
",
                  "codepen_url": "",
                  "type": "code-block",
                },
              ],
              "origin": Object {
                "label": "United Kingdom",
                "value": "GB",
              },
              "part_of_speech": "noun",
              "pronunciation": "web~develop~ment",
              "syllables": Object {
                "count": "5",
                "list": Array [
                  Object {
                    "item": "web",
                  },
                  Object {
                    "item": "de",
                  },
                  Object {
                    "item": "ve",
                  },
                  Object {
                    "item": "lop",
                  },
                  Object {
                    "item": "ment",
                  },
                ],
              },
              "synonyms": Array [
                Object {
                  "post_name": "graphic-design",
                  "post_status": "publish",
                  "post_title": "Graphic Design",
                },
              ],
            },
            "slug": "web-development",
            "terms": Array [
              Object {
                "name": "CSS",
                "slug": "css",
                "taxonomy": "word_tag",
              },
              Object {
                "name": "HTML5",
                "slug": "html5",
                "taxonomy": "word_tag",
              },
              Object {
                "name": "Web Development",
                "slug": "web-development",
                "taxonomy": "word_category",
              },
            ],
            "title": "Web Development 2",
          },
        ]
      }
    />
  </main>
</div>
`;
